ga4gh/datamodel/reads.py

```diff
@@ -209,8 +209,13 @@ def openFile(self, dataFile):
         # not throw an error if the index is missing.
         if not os.path.exists(self._indexFile):
             raise exceptions.FileOpenFailedException(self._indexFile)
-        return pysam.AlignmentFile(
-            self._dataUrl, filepath_index=self._indexFile)
+        try:
+            return pysam.AlignmentFile(
+                self._dataUrl, filepath_index=self._indexFile)
+        except IOError:
+            # IOError thrown when the index file passed in is not actually
+            # an index file... may also happen in other cases?
+            raise exceptions.DataException()
```

Comment:
There should be a file and index name passed to the exception here so we can make a sensible error message. Is there any useful information encoded in the IOError exception thrown back? It might make sense to pass this along as a supllemental message as well.